src/main/java/com/sap/cap/cds/intellij/lsp/ServerDescriptor.java

```diff
@@ -19,17 +23,62 @@
 public class ServerDescriptor extends ProjectWideLspServerDescriptor {
 
     private static final String RELATIVE_SERVER_PATH = "cds-lsp/node_modules/@sap/cds-lsp/dist/main.js";
+    private static final String RELATIVE_SERVER_PKG_PATH = "cds-lsp/node_modules/@sap/cds-lsp/package.json";
     private static final String RELATIVE_MITM_PATH = "cds-lsp/mitm.js";
     private static final String RELATIVE_LOG_PATH = "cds-lsp/stdio.log";
+    public static final int SERVER_EXIT_NODE_VERSION = 72;
 
     public ServerDescriptor(@NotNull Project project, @NotNull String presentableName) {
         super(project, presentableName);
     }
 
     @NotNull
     @Override
-    public GeneralCommandLine createCommandLine() throws ExecutionException {
-        return (isDebugCdsLsp() ? getDebugCommandLine() : getCommandLine())
+    public OSProcessHandler startServerProcess() throws ExecutionException {
+        OSProcessHandler handler = new OSProcessHandler(getCommandLine()) {
+            @Override
+            protected BaseOutputReader.@NotNull Options readerOptions() {
+                return BaseOutputReader.Options.forMostlySilentProcess();
+            }
+        };
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) { /*ignore*/ }
+
+        Process process = handler.getProcess();
+        int exitValue;
+        try {
+            exitValue = process.exitValue();
+        } catch (RuntimeException e) {
+            // Process is still running
+            return handler;
+        }
+        handleServerError(exitValue);
+        return handler;
+    }
+
+    private void handleServerError(int exitValue) {
+        if (exitValue == SERVER_EXIT_NODE_VERSION) {
+            String version;
+            try {
+                version = new String(Files.readAllBytes(Paths.get(resolve(RELATIVE_SERVER_PKG_PATH))));
+                int versionStart = version.indexOf("\"node\": \"") + 9;
```

Comment:
Brittle: This relies on a default formatted `package.json`. Plus if there should ever be e.g. a custom config node with an element `"node": ` and this config node would be above the engine `node` node, it would use wrong values. (I know this is unlikely)

Isn't there a JSON reader in Java these days?

Reply:
Well, I know; just wanted to keep it simple. But maybe we should go for a little more stability here indeed.